### ISSUE TITLE
question about loss function

### ISSUE THREAD

**Issue by Ray0089** [Ending2015a/liepose-diffusion#3]
Thanks for your great work! 
My question is about the loss function of 'chordal_distance' which measure the eliment-wise error of the transformation matrix. Intuitively, it seems that this approach would perform worse than using geodesic distance. Could you please share the reasons for using 'chordal_distance' as the loss function?

**Comment by Ending2015a** [comment]
Hello @Ray0089,
Both chordal and geodesic distance are valid distance metrics on SO(3) and SE(3), and both can work well.
Please refer to the following resources:
* [Rotation Averaging, sec 4](https://users.cecs.anu.edu.au/~hartley/Papers/PDF/Hartley-Trumpf:Rotation-averaging:IJCV.pdf)
* [Partial Bi-Invariance of SE(3) Metrics, A Survey of Metrics](https://rpk.lcsr.jhu.edu/wp-content/uploads/2017/08/Partial-Bi-Invariance-of-SE3-Metrics1.pdf)
* [On the Continuity of Rotation Representations in Neural Networks](https://arxiv.org/abs/1812.07035)

We found that using chordal distance during training is more stable and less sensitive to numerical errors, while geodesic distance is more sensitive and the gradient is unstable in the edge cases (e.g. values near pi).
Note that chordal distance shares similarities with the [6D rotation representation](https://arxiv.org/abs/1812.07035), as both uses matrix representationsto calculate the distances in the embedding space but with a different number of columns: the latter uses the first two columns of the rotation matrices, while the former uses all three columns.
